### DISCUSSION TITLE
Create a GatingStrategy from GatingML Document - Lack of Gate Name Function

### DISCUSSION THREAD

**Discussion by Paul119** [whitews/FlowKit#174] [General]
I am using instruction from here: https://flowkit.readthedocs.io/en/latest/notebooks/flowkit-tutorial-part03-gating-strategy-and-gating-results-classes.html#GatingResults-Class because I have GatingML xml document. 
I've tried to use g_strat.get_gate_ids() to get gate name, but this function is providing me 'gating:id' ("Gate_123456_blablabla") from xml rather than 'name' like "Lymphoid cells".

Is there any function in FlowKit which will provide me correct value for gate name?

**Comment by whitews** [comment]
Hi Paul,

The GatingML specification uses the `gating:id` attribute to identify gates. Other attributes such as `name` can be added by those writing GML docs, but it is not part of the specification. Allowing these extra attributes to pass through to the Gate objects in FlowKit would take some time. I'm curious what software / system generated the file?

**Reply by Paul119** [reply]
We've got the GatingML xml export file from Cytobank: https://developer.cytobank.org/#57374a8a-1373-4069-8e6f-9443ab828968
How I will know that you've added this new function? I will receive any alert/message?

**Reply by whitews** [reply]
I don't have definite plans, but if this functionality is added it would appear in the [release change logs](https://github.com/whitews/FlowKit/releases). Just speculating, but I would guess they are doing this because the gating IDs are required to be unique, yet population names can be repeated in different gate tree branches.

**Reply by Paul119** [reply]
ok, thanks for the info about your releases. 
Yes, population names can be repeated, but I was planning to add them some custom ID in that case rather than using unique gating ID as they are not so meaningful.
